avocado/utils/configure_network.py

```diff
@@ -35,111 +35,176 @@ class NWException(Exception):
         except process.CmdError as ex:
-            raise NWException("ifup fails: %s" % ex)
+            raise NWException("MTU size can not be set: %s" % ex)
+        try:
+            cmd = "ip add show %s" % self._interface
+            mtuvalue = process.system_output(cmd, shell=True).decode("utf-8") \
+                .split()[4]
+            if mtuvalue == mtu:
+                return True
+        except Exception as ex:  # pylint: disable=W0703
+            raise NWException("setting MTU value in host failed: %s", ex)
+        return False
 
-    if distro.detect().name == 'SuSE':
-        conf_file = "/etc/sysconfig/network/ifcfg-%s" % interface
-        if os.path.exists(conf_file):
-            shutil.move(conf_file, conf_file+".backup")
-        else:
-            raise NWException("%s interface not available" % interface)
-        with open(conf_file, "w") as network_conf:
-            network_conf.write("IPADDR=%s \n" % ipaddr)
-            network_conf.write("NETMASK='%s' \n" % netmask)
-            network_conf.write("IPV6INIT=yes \n")
-            network_conf.write("IPV6_AUTOCONF=yes \n")
-            network_conf.write("IPV6_DEFROUTE=yes")
-
-        cmd = "ifup %s" % interface
+    def is_interface_link_up(self):
+        """
+        Checks if the interface link is up
+        :return: True if the interface's link comes up, False otherwise.
+        """
+        if 'up' in genio.read_file("/sys/class/net/%s/operstate" % self._interface):
+            log.info("Interface %s link is up", self._interface)
+            return True
+        return False
+
+    def interface_up(self):
+        cmd = "ifup %s" % self._interface
         try:
             process.system(cmd, ignore_status=False, sudo=True)
             return True
         except process.CmdError as ex:
             raise NWException("ifup fails: %s" % ex)
 
+    def interface_down(self):
+        cmd = "ifdown %s" % self._interface
+        try:
+            process.system(cmd, sudo=True)
+        except Exception as ex:
+            raise NWException("ifdown fails: %s" % ex)
 
-def unset_ip(interface):
-    """
-    Gets interface name unassigns the IP to the interface
-    """
-    if distro.detect().name == 'rhel':
-        conf_file = "/etc/sysconfig/network-scripts/ifcfg-%s" % interface
-
-    if distro.detect().name == 'SuSE':
-        conf_file = "/etc/sysconfig/network/ifcfg-%s" % interface
+    def _move_interfaceconfile(self, src_conf, dest_conf):
+        if os.path.exists(src_conf):
+            shutil.move(src_conf, dest_conf)
+        else:
+            raise NWException("%s interface not available" % self._interface)
 
-    cmd = "ifdown %s" % interface
-    try:
-        process.system(cmd, sudo=True)
-        shutil.move(conf_file+".backup", conf_file)
-        return True
-    except Exception as ex:
-        raise NWException("ifdown fails: %s" % ex)
+    def get_hwaddr(self):
+        f = open('/sys/class/net/%s/address' % self._interface)
```

Comment:
```python3
with open('/sys/class/net/%s/address' % self._interface) as fp:
    return fp.read().strip()
```

Reply:
but also would be nice to put inside a try... except in case interface do not exists.

Reply:
yes make sense